src/hotspot/share/services/memTracker.hpp

```diff
@@ -103,13 +103,15 @@ class MemTracker : AllStatic {
   }
 
   // Record malloc free and return malloc base address
-  static inline void* record_free(void* memblock) {
+  static inline void* record_free_block(void* memblock) {
     // Never turned on
     assert(memblock != NULL, "caller should handle NULL");
-    if (!enabled()) {
-      return memblock;
-    }
-    return MallocTracker::record_free(memblock);
+    assert(enabled(), "NMT must be enabled");
+    return MallocTracker::record_free_block(memblock);
+  }
+  static inline void record_free(FreePackage free_package) {
+    assert(enabled(), "NMT must be enabled");
+    MallocTracker::record_free(free_package);
```

Comment:
Please leave the `enabled()` checks down in `MemTracker`. You rob it of its purpose in life :) Seriously, should we want to revise that, we should do it for all MemTracker methods.